parser_test.go

```diff
@@ -275,6 +78,23 @@ func TestParserDoesNotValidateInvalidPaths(t *testing.T) {
 	}
 }
 
+func TestParserThrowsPanicWhenExpressionIsInvalid(t *testing.T) {
+	defer func() {
+		if r := recover(); r == nil {
+			t.Errorf("The code did not panic")
```

Comment:
good catch, but this makes me feel like refactor asap the panics and move to errors. I will prioritise this task

Reply:
Yes, you are right